addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -311,11 +322,15 @@ def _get_l10n_in_edi_line_details(self, index, line, line_tax_details, sign):
             "Unit": line.product_uom_id.l10n_in_code and line.product_uom_id.l10n_in_code.split("-")[0] or "OTH",
             # Unit price in company currency and tax excluded so its different then price_unit
             "UnitPrice": self._l10n_in_round_value(
-                ((line.balance / (1 - (line.discount / 100))) / line.quantity) * sign, 3),
+                unit_price_in_inr if full_discount_or_zero_quantity else (
+                ((line.balance / (1 - (line.discount / 100))) / line.quantity) * sign), 3),
```

Comment:
Why not have a variable undiscounted_balance?  @jva-odoo 

And maybe there is a way to improve the indentation to improve readability?

Reply:
I think we can also use this unit_price_in_inr like
```
        unit_price_in_inr = 0.00
        full_discount_or_zero_quantity = line.discount == 100.00 or float_is_zero(line.quantity, 3)
        if full_discount_or_zero_quantity:
            unit_price_in_inr = line.currency_id._convert(
                line.price_unit,
                line.company_currency_id,
                line.company_id,
                line.date or fields.Date.context_today(self)
                )
        else:
            unit_price_in_inr = ((line.balance / (1 - (line.discount / 100))) / line.quantity) * sign
```